hydra-node/src/Hydra/Chain/Direct/Tx.hs

```diff
@@ -546,13 +546,16 @@ abortTx scriptRegistry vk (headInput, initialHeadOutput, ScriptDatumForTxIn -> h
     toScriptData (Commit.redeemer Commit.ViaAbort)
 
   reimbursedOutputs =
-    mapMaybe (mkCommitOutput . snd) $ Map.elems commitsToAbort
+    fmap snd $
+      sortOn
+        fst
+        $ mapMaybe (mkCommitOutput . snd) $ Map.elems commitsToAbort
 
-  mkCommitOutput :: ScriptData -> Maybe (TxOut CtxTx)
+  mkCommitOutput :: ScriptData -> Maybe (TxIn, TxOut CtxTx)
   mkCommitOutput x =
     case fromData @Commit.DatumType $ toPlutusData x of
       Just (_party, _validatorHash, serialisedTxOut, _headId) ->
-        toTxContext <$> convertTxOut serialisedTxOut
+        second toTxContext <$> convertTxOut serialisedTxOut
       Nothing -> error "Invalid Commit datum"
```

Comment:
I see that this approach is consistent to what we did here, but I think we should rather be using the `UTxO` we observed from the commits through the `HeadLogic`, i.e. just use that UTxO set as `reimbursedOutputs = Map.elems utxo`. This should work (!) because the ordering of outputs should be canonical.

Reply:
Or `snd <$> UTxO.pairs` .. essentially taking all the `TxOut` values in the order of their `TxIn` keys.

Reply:
Did try this in and pushed with https://github.com/input-output-hk/hydra/commit/168fd15570ad3a00e8c0c07dc081afa5167f715b.